form-regex.rst

```diff
@@ -0,0 +1,91 @@
+3. **Quantification**: A quantifier after a token (such as a character) or group specifies how often that preceding element is allowed to occur. 
+
+    - Square brackets indicate the occurrence of elements within. For example, ``[abc]`` matches either a, b or c character, ``[a-g]`` matches any character from a to g, ``[^abc]`` matches any character except a,b or c.
+    - The question mark indicates zero or one occurrence of the preceding element. For example, ``colou?r`` matches both color and colour.
+    - The asterisk indicates zero or more occurrences of the preceding element. For example, ``ab*c`` matches ac, abc, abbc, abbbc, and so on.
+    - The plus sign indicates one or more occurrences of the preceding element. For example, ``ab+c`` matches abc, abbc, abbbc, and so on, but not ac.
+    - ``{n}``: The preceding item is matched exactly n times. 
+    - ``{min,}``:	The preceding item is matched min or more times. 
+    - ``{min,max}``: The preceding item is matched at least min times, but not more than max times. 
+
+
+For a clear understanding of regular expressions, try these regex online checker tools: https://regex101.com/, https://www.regextester.com/, https://regexr.com/, http://www.regexe.com/.
+
+.. _tips-on-regex:
+
+Tips on using regular expressions
+----------------------------------
+
+- Regular expressions only apply to **strings**. The function used is of the form: ``regex(string value, string expression)`` where it returns the result of regex test on provided value. 
+
+- Be careful while setting limits on length of a constraint.
+  For example, if you want to restrict an input to a maximum of six alphabets, you can do so as follows:
+
+ In **XLSForm**:
+
+ .. csv-table:: **Survey**
+   :header: "type", "name", "label", "constraint", "constraint_message"
+   :widths: auto
+
+   "text", "ex", "Enter example", "regex(.,'^[a-zA-Z]{0,6}$')", "Input can have a maximum of 6 characters."
+
+ In **XForm XML**:
+
+ .. code-block:: xml
+
+   <bind constraint="Regex(.,'^[a-zA-Z]{0,6}$')" jr:constraintMsg="Input can have a maximum of 6 characters." nodeset="/regex_ex/ex" type="string"/>
+
+   <input ref="/regex_ex/ex">
+      <label>Enter example</label>
+    </input>
+
+ Instead if you used a constraint of the form ``regex(.,'^[a-zA-Z]{6}$')``, it would require an input of exactly six characters.
+
+ .. note::
+
+   ``^`` and ``$`` denote start and end of string respectively.
+
+- If you want to use a regular expression constraint on a number, first of all, make sure that the type of your question is **text** and appearance is **numbers** and then apply the constraint. 
+
+ For example, to restrict the telephone numbers to have exactly 10 digits:
+
+ In **XLSForm**:
+
+ .. csv-table:: **Survey**
+   :header: "type", "name", "label", "constraint", "constraint_message", "appearance"
+   :widths: auto
+
+   "text", "tel_no", "Enter your Telephone number", "regex(.,'^[0-9]{10}$')", "Telephone numbers should have 10 digits", "numbers"
```

Comment:
This is a weird example, because tel numbers often have parens, hyphens, dots, or other punctuation when written in a standard format.

(Check out https://gist.github.com/nerdsrescueme/1237767 and other similar lists for common regex patterns.)